app/presenters/hyrax/fixity_status_presenter.rb

```diff
@@ -0,0 +1,89 @@
+module Hyrax
+  # Creates fixity status messages to display to user, for a fileset.
+  # Determines status by looking up existing recorded ChecksumAuditLog objects,
+  # does not actually do a check itself.
+  #
+  # See FileSetFixityCheckService and ChecksumAuditLog for actually performing
+  # checks and recording as ChecksumAuditLog objects.
+  class FixityStatusPresenter
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TextHelper
+    include ActionView::Helpers::OutputSafetyHelper
+
+    attr_reader :file_set_id, :relevant_log_records
+    # Note this takes a file_set_id NOT a FileSet, easy use from either solr
+    # or AF object.
+    def initialize(file_set_id)
+      @file_set_id = file_set_id
+    end
+
+    # Returns a html_safe string communicating fixity status checks,
+    # possibly on multiple files/versions.
+    def render_file_set_status
+      @file_set_status ||=
+        if relevant_log_records.empty?
+          "Fixity checks have not yet been run on this object"
```

Comment:
There are some strings in this PR that I'd like to see handled via i18n, but this PR is big enough and has waited long enough that it's reasonable for this to be ticketed and worked on subsequently IMO.